providers/base/bin/wifi_nmcli_test.py

```
@@ -512,9 +497,18 @@ def main():
         finally:
             turn_up_connection(activated_uuid)
             delete_test_ap_ssid_connection()
-            restore_netplan_files(str(temp_dir.name), NETPLAN_DIR)
-            # cannot use temp_dir.cleanup to support old pythons
-            cleanup_netplan_backup(str(temp_dir.name))
+
+
+def main():
+    temp_dir = tempfile.TemporaryDirectory()
+    backup_netplan_files(str(temp_dir.name), NETPLAN_DIR)
+    try:
+        run()
+    except Exception:
+        restore_netplan_files(str(temp_dir.name), NETPLAN_DIR)
+        raise
+
+    restore_netplan_files(str(temp_dir.name), NETPLAN_DIR)
```

I might be wrong, but I think this can be simplified to:

```python
   try:
        run()
    finally:
        restore_netplan_files(str(temp_dir.name), NETPLAN_DIR)
```

---

Are you sure the exception will be lost?

I just tried this:

```python
>>> def run():
...     raise NotImplementedError("Oops!")
... 
>>> try:
...     run()
... finally:
...     print("This will be printed regardless")
... 
This will be printed regardless
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
  File "<stdin>", line 2, in run
NotImplementedError: Oops!
```

---

(plus, [flake8 doesn't like the bare except statement!)](https://www.flake8rules.com/rules/E722.html)

---

Well, I don't remember how I tested that, but your test seems perfectly fine to me. Committed the changes.